src/sidebar/groups.js

```diff
@@ -81,22 +143,36 @@ function groups(localStorage, serviceUrl, session, $rootScope, store) {
       }
     }
   }
-
+  
   // reset the focused group if the user leaves it
   $rootScope.$on(events.GROUPS_CHANGED, function () {
-    if (focusedGroup) {
-      focusedGroup = get(focusedGroup.id);
-      if (!focusedGroup) {
-        $rootScope.$broadcast(events.GROUP_FOCUSED, focused());
+    // FIXME A better way of doing this?
+    load();
+  });
+
+  // refetch groups when user changes
+  $rootScope.$on(events.USER_CHANGED, () => {
+    // FIXME Makes a second api call on page load. better way?
+    load();
```

Comment:
It may be helpful to draw a flow diagram / state machine here of the desired flow.

In doing this, I think you'll find that some information is missing from this event. It is fired when the profile is (1) _initially fetched_, as well as when the (2) user logs in or (3) logs out. You only want to re-fetch groups however when (2) or (3) happen.